### PR TITLE
only use arguments on suda#executable if it's equal to "sudo"

### PR DESCRIPTION
**Pull request by aarondill**
This ensures that the sudo options work correctly, and creates *rudimentary* support for other commands (such as `doas`, see #40)

This isn't perfect, but likely the best we can do (see [my comment](https://github.com/lambdalisue/suda.vim/issues/40#issuecomment-1640903171)).

This change allows any command which accepts no password, or else a password from stdin to work. Any command set to `g:suda#executable` will be executed with no further arguments (so the user may need to provide any needed), and will be run, first with an empty stdin, then with the user's provided password as stdin. Further support would likely need to be added on a case-by-case basis and may quickly spiral out of control. 

## Summary by CodeRabbit

- **New Features**
	- Enhanced command construction for improved reliability and security.
	- Introduced interactive password prompts for certain operations.

### CODE REVIEW

**Comment by acid-bong** [comment]
Just tested, it doesn't pass the password to doas, and 2 failed attempts it somehow counted as 3

**Comment by aarondill** [comment]
The password is passed through stdin, but `doas` doesn't process it. Also, it is called twice, one without stdin, once with (as noted above).

This will *only* work with `doas` when the user has the `nopass` privilege.

**Comment by aarondill** [comment]
One tool that *does* work with this change is `su - -c` (as su accepts input through stdin), though I believe a wrapper would be required to ensure that all arguments passed end up as one after the `-c` (like the one below)
```
#!/usr/bin/env sh
exec su - -c "$*"
```

**Comment by acid-bong** [comment]
The difference is that `su` asks for root's password, not your user's, like sudo and doas do

**Comment by aarondill** [comment]
> The difference is that `su` asks for root's password, not your user's, like sudo and doas do

This is true, but as I've already said, unless you can discover a way to pass the password to doas, we can only support any tools that accept the password through stdin (like, `sudo` and `su`)

**Comment by lambdalisue** [comment]
My apologies for the delay. I completely overlooked this matter. Could you please resolve the conflicts first so we can proceed with the review? @aarondill

**Comment by aarondill** [comment]
@lambdalisue I've rebased this onto HEAD. 
I've also introduced a few changes:
The checking introduced in https://github.com/lambdalisue/suda.vim/commit/257767d7977414210b6db9df0be39b9789d83978 is *only* done on 'sudo', since there's no guarantee that the user provided command will ever support not passing a password. Additionally, the user provided command is called the same way either way: `user_sudo command` with no options.

File: autoload/suda.vim
Comment (aarondill) [reply autoload/suda.vim:10]: While I personally think we should pass `--` (at least for `sudo`), this isn't present in the original source, so I'll leave this up to you, @lambdalisue.

As to the other 2 TODOs that I left, I'd like to implement these, if you'd allow me to; however, they are both out of scope of this PR. I'll implement them after this PR is merged, if you'd merge it.

**Review by lambdalisue — APPROVED** [review]
LGTM

**Comment by lambdalisue** [comment]
Thanks a lot 🎉

**Pull request merged**
